### PR TITLE
Add i64 tensor argument support and bring back GatherModule_basic

### LINKED ISSUE #324: i64 tensor argument support for refbackend
`GatherModule_basic`  need this to pass. See https://github.com/llvm/mlir-npcomp/pull/323.

### PR DESCRIPTION
**Pull request by cathyzhyi**
Fix #324

### CODE REVIEW

**Comment by silvasean** [comment]
I'm not sure that our usual xfail error catching can catch this error successfully.

You can add a try/catch around this line https://github.com/llvm/torch-mlir/blob/2b18aad8073d906f76da72571c32ab015dd5974b/python/torch_support/npcomp_torchscript/e2e_test/framework.py#L310 and then add a runtime_error field to TestResult that can hold the stringified exception.

**Comment by cathyzhyi** [comment]
Interesting. When running this locally the error is raised at compile time `argument must be a memref of f32` in `MungeCallingConventions` rather than at invoking.

**Comment by silvasean** [comment]
oh, then we are probably jsut missing error handing here: https://github.com/llvm/torch-mlir/blob/e291aa688f708d176edb8d3950196f6f28dd6e2e/python/npcomp/compiler/pytorch/backend/refbackend.py#L103

**Comment by cathyzhyi** [comment]
I am seeing the same CI failure (`assert result is not None, "Invocation didn't produce a result"`) in https://github.com/llvm/torch-mlir/pull/326 which doesn't have the `GatherModule_basic ` enabled. I can't reproduce this failure locally though.

**Comment by silvasean** [comment]
I think if you add the code to catch the runtime errors, you should be able to at least see which test in the CI is failing.

File: python/torch_support/npcomp_torchscript/e2e_test/framework.py [hunk 1]
```diff
@@ -307,7 +307,16 @@ def run_tests(tests: List[Test], config: TestConfig) -> List[TestResult]:
                            golden_trace=None))
             continue
         # TODO: Run in parallel.
-        trace = config.run(compiled, golden_trace)
+        try:
+            trace = config.run(compiled, golden_trace)
+        except Exception as e:
+            results.append(
+                TestResult(unique_name=test.unique_name,
+                           compilation_error=str(e),
```
Comment (silvasean) [inline python/torch_support/npcomp_torchscript/e2e_test/framework.py:315]: please add a new field runtime_error to hold this, and a new test python/test/torchscript_e2e_test/runtime_failure.py

**Review by silvasean — APPROVED** [review]

**Comment by silvasean** [comment]
Thanks for fixing this!

File: python/torch_support/npcomp_torchscript/e2e_test/framework.py (hunk 1 above)
Comment (cathyzhyi) [reply python/torch_support/npcomp_torchscript/e2e_test/framework.py:315]: got it!

**Comment by silvasean** [comment]
Interesting. That `SystemError: null argument to internal routine` was the same issue that MmModule_chained had when I was debugging it. This might be the same issue. Not sure if you want to debug it, but it is probably something we need to fix for the refbackend anyway. For now, we can XFAIL MmModule_basic too (and uncomment MmModule_chained now that we can properly XFAIL it).

**Comment by cathyzhyi** [comment]
Seems it's always the first test failing. If I comment out both `MmModule_basic` and `MmModule_chained` the nexxt one `BmmModule_basic` would fail. I am leaving `MmModule_basic` as the only xfail for now.

**Pull request merged**